pkg/webhook/operatingsystemmanager/operatingsystemconfig/validation/validation_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+
+	osmv1alpha1 "k8c.io/operating-system-manager/pkg/crd/osm/v1alpha1"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var (
+	testScheme = runtime.NewScheme()
+)
+
+func init() {
+	_ = osmv1alpha1.AddToScheme(testScheme)
+}
+
+// TestHandle tests the admission handler
+func TestHandle(t *testing.T) {
+	osc := getOperatingSystemConfig()
+	oscRaw := oscToRawExt(osc)
+
+	osc.Spec.OSVersion = "fake"
+	oscRawUpdate := oscToRawExt(osc)
+
+	tests := []struct {
+		name        string
+		req         webhook.AdmissionRequest
+		wantAllowed bool
+	}{
+		{
+			name: "Delete osc success",
+			req: webhook.AdmissionRequest{
+				AdmissionRequest: admissionv1.AdmissionRequest{
+					Operation: admissionv1.Delete,
+					RequestKind: &metav1.GroupVersionKind{
+						Group:   osmv1alpha1.GroupName,
+						Version: osmv1alpha1.GroupVersion,
+						Kind:    "OperatingSystemConfig",
+					},
+					Name: "osc",
+				},
+			},
+			wantAllowed: true,
+		},
+		{
+			name: "Create osc success",
+			req: webhook.AdmissionRequest{
+				AdmissionRequest: admissionv1.AdmissionRequest{
+					Operation: admissionv1.Create,
+					RequestKind: &metav1.GroupVersionKind{
+						Group:   osmv1alpha1.GroupName,
+						Version: osmv1alpha1.GroupVersion,
+						Kind:    "OperatingSystemConfig",
+					},
+					Name:   "osc",
+					Object: oscRaw,
+				},
+			},
+			wantAllowed: true,
+		},
+		{
+			name: "Update osc success",
```

Comment:
Isn't this test case used to verify that it's going to be rejected? If yes, I would name this something like:

```suggestion
			name: "Update osc rejected",
```